events/api/src/test/java/org/projectnessie/events/api/TestJsonSerde.java

```diff
@@ -0,0 +1,373 @@
+                    .id("id")
+                    .snapshotId(1L)
+                    .schemaId(2)
+                    .specId(3)
+                    .sortOrderId(4)
+                    .build())
+            .build();
+    assertThat(deserialize(serialize(event), Event.class)).isEqualTo(event);
+  }
+
+  @Test
+  void contentRemoved() throws Exception {
+    ContentRemovedEvent event =
+        ImmutableContentRemovedEvent.builder()
+            .branch("branch1")
+            .hash("hash1")
+            .contentKey(ContentKey.of("ns", "table1"))
+            .id(UUID.randomUUID())
+            .repositoryId("repo1")
+            .createdAt(Instant.now())
+            .createdBy("Alice")
+            .sentBy("Nessie")
+            .build();
+    assertThat(deserialize(serialize(event), Event.class)).isEqualTo(event);
+  }
+
+  @Test
+  void icebergTable() throws Exception {
+    IcebergTable content =
+        ImmutableIcebergTable.builder()
+            .id("id")
+            .metadataLocation("metadataLocation")
+            .snapshotId(1L)
+            .schemaId(2)
+            .specId(3)
+            .sortOrderId(4)
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .build();
+    assertThat(deserialize(serialize(content), Content.class)).isEqualTo(content);
+  }
+
+  @Test
+  void deltaLakeTable() throws Exception {
+    DeltaLakeTable content =
+        ImmutableDeltaLakeTable.builder()
+            .id("id")
+            .addCheckpointLocationHistory("checkpoint")
+            .addMetadataLocationHistory("metadata")
+            .lastCheckpoint("lastCheckpoint")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .build();
+    assertThat(deserialize(serialize(content), Content.class)).isEqualTo(content);
+  }
+
+  @Test
+  void icebergView() throws Exception {
+    IcebergView content =
+        ImmutableIcebergView.builder()
+            .id("id")
+            .metadataLocation("metadataLocation")
+            .versionId(1L)
+            .schemaId(2)
+            .sqlText("sqlText")
+            .dialect("dialect")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .build();
+    assertThat(deserialize(serialize(content), Content.class)).isEqualTo(content);
+  }
+
+  @Test
+  void customEvent() throws Exception {
+    CustomEvent event =
+        ImmutableCustomEvent.builder()
+            .id(UUID.fromString("7385d1e6-3deb-440b-9008-a383e2de6e6c"))
+            .customType("weird")
+            .repositoryId("repo1")
+            .createdAt(Instant.parse("2023-04-25T13:02:05Z"))
+            .sentBy("Nessie")
+            .putProperty("string", "foo")
+            .putProperty("number", 123)
+            .putProperty("boolean", true)
+            .build();
+    assertThat(deserialize(serialize(event), Event.class)).isEqualTo(event);
+  }
+
+  @Test
+  void customContent() throws Exception {
```

Comment:
nit: maybe add a test for `Namespace` too... or did I miss it?

Reply:
Good catch!